src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaDataResultSet.java

```diff
@@ -414,6 +431,10 @@ public double getDouble(int columnIndex) throws SQLException {
     }
   }
 
+  public String getQueryID() {
```

Comment:
Added a function to retrieve queryID from SnowflakeDatabaseMEtaDataResultSet class. Even if the query is not successful, the ID still exists.